operations-stdlib/src/commonMain/kotlin/string/compareToDate/ComparePrecision.kt

```diff
@@ -0,0 +1,60 @@
+package string.compareToDate
+
+enum class ComparePrecision{
+    MILLISECOND,
+    SECOND,
+    MINUTE,
+    HOUR,
+    DAY,
+    MONTH,
+    YEAR;
+
+    fun position(): Int {
+        return position(this)
+    }
+
+    fun dateSuffix(): String {
+        return dateSuffix(this)
+    }
+
+    companion object {
+        private const val secondPosition = 19
+        private const val minutePosition = 16
+        private const val hourPosition = 13
+        private const val dayPosition = 10
+        private const val monthPosition = 7
+        private const val yearPosition = 4
+        private const val dateLength = 24
+
+        private const val secondsPrecision = ".000Z"
+        private const val minutesPrecision = ":00$secondsPrecision"
+        private const val hoursPrecision = ":00$minutesPrecision"
+
+        private const val daysPrecision = "T00$hoursPrecision"
+        private const val monthsPrecision = "-01$daysPrecision"
+        private const val yearsPrecision = "-01$monthsPrecision"
+        fun dateSuffix(precision: ComparePrecision): String {
```

Comment:
It should not be separate function. This function only passes call to another function - IMO it just adds complexity
```
fun dateSuffix(): String {
        return dateSuffix(this)
    }
```

I also think that this responsibility should be extracted to separate class - so that it can be tested in separation. 
So that way `ComparePrecision` is just a simple enum, and you have another class responsible for preparing formattedDate

So formatDate will be a public function in this class, and dateSuffix(precision: ComparePrecision) and precision(precision: ComparePrecision) will be private.
I think that way responsibilities will be better defined.